miden-lib/asm/scripts/ASWAP.masm

```diff
@@ -0,0 +1,94 @@
+use.miden::sat::note
+use.miden::wallets::basic->wallet
+
+#! Helper procedure to add all assets of a note to an account.
+#!
+#! Inputs: []
+#! Outputs: []
+#!
+proc.add_note_assets_to_account
+    push.0 exec.note::get_assets
+    # => [num_of_assets, 0 = ptr, ...]
+
+    # compute the pointer at which we should stop iterating
+    dup.1 add
+    # => [end_ptr, ptr, ...]
+
+    # pad the stack and move the pointer to the top
+    padw movup.5
+    # => [ptr, 0, 0, 0, 0, end_ptr, ...]
+
+    # compute the loop latch
+    dup dup.6 neq
+    # => [latch, ptr, 0, 0, 0, 0, end_ptr, ...]
+
+    while.true
```

Comment:
can you remove this while loop here? For now we can expect to change 1 full asset for another.

Reply:
done